Python/gc_free_threading.c

```diff
@@ -1878,6 +1901,173 @@ cleanup_worklist(struct worklist *worklist)
     }
 }
 
+// Return the current resident set size (RSS) of the process, in units of KB.
+// Returns -1 if this operation is not supported or on failure.
+static Py_ssize_t
+get_current_rss(void)
+{
+#ifdef _WIN32
+    // Windows implementation using GetProcessMemoryInfo
+    PROCESS_MEMORY_COUNTERS pmc;
+    HANDLE hProcess = GetCurrentProcess();
+    if (NULL == hProcess) {
+        // Should not happen for the current process
+        return -1;
+    }
+
+    // GetProcessMemoryInfo returns non-zero on success
+    if (GetProcessMemoryInfo(hProcess, &pmc, sizeof(pmc))) {
+        // pmc.WorkingSetSize is in bytes. Convert to KB.
+        return (Py_ssize_t)(pmc.WorkingSetSize / 1024);
+    } else {
+        CloseHandle(hProcess);
```

Comment:
Apparently this is not needed as GetCurrentProcess returns a pseudo handle. It doesn't hurt though